common4j/src/main/com/microsoft/identity/common/java/logging/ProductHelper.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.logging;
+
+import com.microsoft.identity.common.java.AuthenticationConstants;
+import com.microsoft.identity.common.java.platform.Device;
+import com.microsoft.identity.common.java.util.StringUtil;
+
+import cz.msebera.httpclient.android.BuildConfig;
+import lombok.NonNull;
+
+/**
+ * Simple helper class to pull product telemetry information from diagnostic context.
+ */
+public class ProductHelper {
+    private static final String TAG = ProductHelper.class.getSimpleName();
+
+    /**
+     * The String to be returned if the value is not set.
+     */
+    protected static final String NOT_SET = "NOT_SET";
+
+    /**
+     * Returns the product (library) name by accessing it from DiagnosticContext.
+     * @return product name from DiagnosticContext.
+     */
+    @NonNull
+    public static String getProduct() {
+        final String methodName = ":getProduct";
+
+        final String product = DiagnosticContext.INSTANCE.getRequestContext().get(AuthenticationConstants.SdkPlatformFields.PRODUCT);
```

Comment:
maybe do this like the device class?

force set it in the platform component class *once*, under the sGlobalStateInitalized block.

Reply:
corr id makes sense since we need to print that in every log line, and we're not passing the parameter in every place we invoke logging functions.

The library name and library version doesn't belong there. We don't print it in the log. It's not for "diagnostic" purpose (as in "Diagnostic"Context)